src/gui/qgsadvanceddigitizingdockwidget.cpp

```diff
@@ -716,14 +716,21 @@ void QgsAdvancedDigitizingDockWidget::lockAdditionalConstraint( AdditionalConstr
 void QgsAdvancedDigitizingDockWidget::updateCapacity( bool updateUIwithoutChange )
 {
   CadCapacities newCapacities = CadCapacities();
+  const bool isGeographic = mMapCanvas->mapSettings().destinationCrs().isGeographic();
```

Comment:
When you switch to a projected CRS, angle/distance stay disabled.
I guess that a signal should be added when changing layers.

Reply:
I can't reproduce that. If I'm mid-way through drawing a shape and change the crs then the control stays disabled, but also the parts of the shape i've drawn get all distorted anyway because the previous coordinates aren't reprojected automatically. So you're basically forced to cancel the current drawing and re-start if you change the crs mid-way anyway..!

Reply:
I'll go ahead and merge this -- let me know if you do find a way to reproduce your issue and I'll address in a follow up.